src/vm/clsload.cpp

```diff
@@ -100,8 +100,9 @@ PTR_Module ClassLoader::ComputeLoaderModuleWorker(
 
 #ifndef DACCESS_COMPILE
 #ifdef FEATURE_NATIVE_IMAGE_GENERATION
-    // Check we're NGEN'ing
-    if (IsCompilationProcess())
+    // Check we're NGEN'ing. ComputeLoaderModuleForCompilation algorithm assumes that there is fragile
+    // NGen image for CoreLib that is not the case for ReadyToRun compilation.
```

Comment:
NIT: CoreLib always has fragile Ngen image. Shouldn't the comment say something like "ComputeLoaderModuleForCompilation assumes we can save information from fragile CoreLib into R2R image being generated, which is not version resilient incase we wish to mix and match Check corelib with R2R images generated against release CoreLib. Hence, we skip sving such information for R2R" ?